src/components/attribute/EmptyState.tsx

```diff
@@ -0,0 +1,155 @@
+import { useClipboard } from '@aha-app/aha-develop-react';
+import { IDENTIFIER } from '@lib/extension';
+import { linkMergeRequestToRecord } from '@lib/fields';
+import { withGitLabApi } from '@lib/gitlab/api';
+import { getMRFromURL } from '@lib/gitlab/getMRFromURL';
+import { validateMrURL } from '@lib/gitlab/validateMrURL';
+import React, { useEffect, useState } from 'react';
+
+const ICON = aha.iconForExtensionIdentifier(IDENTIFIER);
+const LEARN_MORE_URL = 'https://github.com/aha-develop/gitlab';
+
+type MenuProps = {
+  record: Aha.ApplicationModel;
+  onPaste: Function;
+};
+
+const Menu = ({ record, onPaste }: MenuProps) => {
+  const handleSync = () => {
+    aha.command(`${IDENTIFIER}.sync`, { record });
+  };
+
+  return (
+    <aha-menu>
+      <aha-button slot="button" type="attribute" size="mini">
+        <aha-icon icon="fa-solid fa-ellipsis"></aha-icon>
+      </aha-button>
+      <aha-menu-item onClick={() => onPaste()}>Paste MR link</aha-menu-item>
+      <aha-menu-item onClick={handleSync}>Scan GitLab</aha-menu-item>
+      <hr />
+      <aha-menu-item>
+        <a href={LEARN_MORE_URL} target="_blank" rel="noopener noreferrer">
+          <aha-icon icon="fa fa-external-link" />
+          Read the docs
+        </a>
+      </aha-menu-item>
+    </aha-menu>
+  );
+};
+
+const EmptyStateBox = ({ children }) => (
+  <aha-box class="m-0" style={{ color: 'var(--theme-secondary-text)' }}>
+    <div style={{ margin: 'calc(-2em + 12px)' }}>{children}</div>
+  </aha-box>
+);
+
+export const EmptyState: React.FC<{ record: Aha.RecordUnion }> = ({ record }) => {
+  const [onCopy, copied] = useClipboard();
+  const [pasteMode, setPasteMode] = useState<Boolean>(false);
+  const [validation, setValidation] = useState<String | null>(null);
+  const [hasConfiguredWebhook, setHasConfiguredWebhook] = useState<Boolean | null>(null);
+
+  // aha-menu gets mad if you remove it from the DOM while the menu is open.
+  // Wait 1 tick so the menu has a chance to unmount first.
+  const viewPasteMode = () => {
+    setTimeout(() => setPasteMode(true), 1);
+  };
+
+  const pasteLink = async (mrUrl: string) => {
+    if (!validateMrURL(mrUrl)) {
```

Comment:
If I put in a junk string that's not a URL, the `new URL` constructor will throw, so `validation` doesn't get set. Might be worth a `try/catch` in `validateMrURL()`?